django/db/models/manager.py

```diff
@@ -242,12 +244,40 @@ class Manager(BaseManager.from_queryset(QuerySet)):
 class ManagerDescriptor(object):
     # This class ensures managers aren't accessible via model instances.
     # For example, Poll.objects works, but poll_obj.objects raises AttributeError.
-    def __init__(self, manager):
+    # It also ensures managers defined on non-abstract base classes are not inherited by child classes.
+    def __init__(self, manager, model):
         self.manager = manager
+        self.model = model
 
     def __get__(self, instance, cls=None):
         if instance is not None:
             raise AttributeError("Manager isn't accessible via %s instances" % cls.__name__)
+
+        if cls._meta.concrete_model == self.model._meta.concrete_model:
```

Comment:
@ar45, thank you for comments. I examined the django code a little more and changed the approach to make it less complicated and to use existing django code more properly. I guess that would be enough to work well and it's easier to understand and to maintain. But if your model is not inherited from `django.db.models.Model` I find it hard to predict it's behaviour. Does new approach fulfils your requirements with custom model?